File: app/src/main/java/com/manrique/trailerstock/model/Venta.java
package com.manrique.trailerstock.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "ventas")
public class Venta {
    @PrimaryKey(autoGenerate = true)
    private int id;

    private long fecha; //
    private double total;
    private String tipoCliente; // "Lista" o "Mayorista"
    private boolean aplicoPromo;
    private Integer promocionId; // ID de la promoción aplicada (nullable)

    public Venta(long fecha, double total, String tipoCliente, boolean aplicoPromo, Integer promocionId) {
        this.fecha = fecha;
        this.total = total;
        this.tipoCliente = tipoCliente;
        this.aplicoPromo = aplicoPromo;
        this.promocionId = promocionId;
    }

    // Getters y Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public long getFecha() {
        return fecha;
    }

    public void setFecha(long fecha) {
        this.fecha = fecha;
    }

    public double getTotal() {
        return total;
    }

    public void setTotal(double total) {
        this.total = total;
    }

    public String getTipoCliente() {
        return tipoCliente;
    }

    public void setTipoCliente(String tipoCliente) {
        this.tipoCliente = tipoCliente;
    }

    public boolean isAplicoPromo() {
        return aplicoPromo;
    }

    public void setAplicoPromo(boolean aplicoPromo) {
        this.aplicoPromo = aplicoPromo;
    }

    public Integer getPromocionId() {
        return promocionId;
    }

    public void setPromocionId(Integer promocionId) {
        this.promocionId = promocionId;
    }
}

File: app/src/main/java/com/manrique/trailerstock/data/local/entities/Venta.kt
package com.manrique.trailerstock.data.local.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.manrique.trailerstock.data.local.entities.EstadoVenta
import com.manrique.trailerstock.data.local.entities.MetodoPago

/**
 * Entity que representa una venta realizada.
 */
@Entity(tableName = "ventas")
data class Venta(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    
    @ColumnInfo(name = "fecha")
    val fecha: Long = System.currentTimeMillis(),
    
    @ColumnInfo(name = "total")
    val total: Double,
    
    @ColumnInfo(name = "tipo_cliente")
    val tipoCliente: String,  // "LISTA" o "MAYORISTA"
    
    @ColumnInfo(name = "metodo_pago")
    val metodoPago: MetodoPago = MetodoPago.EFECTIVO,
    
    @ColumnInfo(name = "numero_venta")
    val numeroVenta: String? = null,
    
    @ColumnInfo(name = "notas")
    val notas: String? = null,
    
    @ColumnInfo(name = "estado")
    val estado: EstadoVenta = EstadoVenta.ACTIVA,
    
    // === Columnas de sincronización ===
    @ColumnInfo(name = "synced_at")
    val syncedAt: Long? = null,  // Timestamp de última sincronización con Firestore
    
    @ColumnInfo(name = "device_id")
    val deviceId: String = "",  // UUID del dispositivo que creó la venta
    
    @ColumnInfo(name = "remote_id")
    val remoteId: String? = null  // ID del documento en Firestore
) {
    companion object {
        const val TIPO_LISTA = "LISTA"
        const val TIPO_MAYORISTA = "MAYORISTA"
    }
}
